Makefile

```diff
@@ -26,6 +26,21 @@ build:
 install:
 	@go install -ldflags='$(LDFLAGS)' ./cmd/...
 
+create-release-artifacts:
+	for os in darwin linux windows; do \
+		zqdir=zq-$(VERSION).$${os}-amd64 ; \
+		rm -rf dist/$${zqdir} ; \
+		mkdir -p dist/$${zqdir} ; \
+		GOOS=$${os} GOARCH=amd64 go build -ldflags='$(LDFLAGS)' -o dist/$${zqdir} ./cmd/... ; \
+	done
+	rm -rf dist/release && mkdir -p dist/release
+	cd dist && for d in zq-$(VERSION)* ; do \
+		zip -r release/$${d}.zip $${d} ; \
+	done
```

Comment:
This isn't a blocker; it's a commentary on the zq Makefile and the direction it's going.

I understand that Go tooling obviates some of the traditional reasons to use make, but it seems like where we can, we could still make traditional use of Makefiles: real, non-phony targets, and dependency chains. If I run `make create-release-artifacts` twice, the work is done twice. In most make user experiences this is not expected.

There's also a bit of `go build` duplication happening here with `make build`.

The way I could see this working with deps conceptually is something like:
```
non-phony-arch-specifc-build:
        go build instructions to build arch-specific-build-target

build:
       phony shortcut for a single arch-specifc-build for the calling arch; possibly creates a symlink from dist/zq to dist/arch/zq

non-phony-arch-release-zip: non-phony-arch-specifc-build
        zip action against the dep to produce the target

all-non-phony-release-zips: individual arch-release zips

create-release-artifacts: all-non-phony-release-zips

publish-release-artifacts: all-non-phony-release-zips
        leave as is, basically

.PHONY: build create-release-artifacts publish-release-artifacts
```
One way around fewer phony targets is to emit stamps via `touch(1)` when the target's work is done. That might be appropriate for `all-non-phony-release-zips` in this example.

This lets you remove the `rm -rf` setup you need (that should be part of `make clean`). A second `make` run then won't reduplicate work it doesn't need to. The `go build` duplication is gone. Finally, it removes a lot of shell from Makefile and looks more Makefile-y.

Reply:
@mikesbrown : thanks for the comments, and the +1 . As you say, since there's a great deal of optimization within go tooling for building packages & caching the results, there's not as much of a need for exposing the source to target dependencies within make.

We have some other Makefile related changes we should do - like incoporating the zql/Makefile work into this top level one. I'll go ahead with this PR for now, but we'll plan on making larger changes & establishing dependency changes in the future.